internal/interface/interface.go

```diff
@@ -0,0 +1,7 @@
+package _interface
```

Comment:
I've not seen this pattern with package names before, should this just be `interface`?

Also, maybe provide the copyright boilerplate and a docstring briefly talking about the package, since it's the only file here?

Reply:
I used `iface` since interface is a keyword.  Also, good catch on copywrite boilerplate.